src/core/tsi/transport_security.c

```diff
@@ -180,16 +182,59 @@ tsi_result tsi_handshaker_create_frame_protector(
   result = self->vtable->create_frame_protector(self, max_protected_frame_size,
                                                 protector);
   if (result == TSI_OK) {
-    self->frame_protector_created = 1;
+    self->frame_protector_created = true;
   }
   return result;
 }
 
+tsi_result tsi_handshaker_next(
+    tsi_handshaker *self, const unsigned char *received_bytes,
+    size_t received_bytes_size, unsigned char **bytes_to_send,
+    size_t *bytes_to_send_size, tsi_handshaker_result **handshaker_result,
+    tsi_handshaker_on_next_done_cb cb, void *user_data) {
+  if (self == NULL) return TSI_INVALID_ARGUMENT;
+  if (self->handshaker_result_created) return TSI_FAILED_PRECONDITION;
+  return self->vtable->next(self, received_bytes, received_bytes_size,
```

Comment:
Please check that next is not NULL (looks like it could for some implementations).

It's actually a good idea to check vtable and the method function pointers in all these helper functions. It would be good to cleanup the others as well. Thanks!